### ISSUE TITLE
writing one compressed image forces subsequent ones to be compressed

### ISSUE THREAD

**Issue by sbailey** [esheldon/fitsio#150]
Writing a compressed image to an open FITS file causes all subsequent images to be compressed.  If the images are written individually using `fitsio.write` then the expected behavior works where individual images are compressed only if requested.  Example:
```python
import numpy as np
import fitsio

x = np.random.uniform(0,1,size=(5,8))

with fitsio.FITS('blat.fits', mode='rw', clobber=True) as fx:
    fx.write(x, extname='A')
    fx.write(x, extname='B', compress='gzip')
    fx.write(x, extname='C', compress=None)  #- OOPS: will be compressed too   

fitsio.write('foo.fits', x, extname='A', clobber=True)
fitsio.write('foo.fits', x, extname='B', compress='gzip')
fitsio.write('foo.fits', x, extname='C', compress=None) #- Correctly will not be compressed

for filename in ['blat.fits', 'foo.fits']:
    print('Reading {}'.format(filename))
    fx = fitsio.FITS(filename)
    for hdu in fx:
        print('  {:8s} {}'.format(hdu.get_extname(), hdu.is_compressed()))
```

Produces:
```
Reading blat.fits
  A        False
  B        True
  C        True
Reading foo.fits
  A        False
  B        True
  C        False
```
I was expecting HDU C of blat.fits to be uncompressed, just like HDU C of foo.fits.

**Comment by esheldon** [comment]
thanks for the report.  Do you have a fix for this in mind?

**Comment by sbailey** [comment]
From a quick read of the code I don't see why it isn't working and was hoping that you would know faster.  i.e. I don't see the code ignoring the `compress` keyword so I don't have a fix in mind without needing to do more detailed debugging.

**Comment by esheldon** [comment]
@sbailey I have a fix for this in PR #218
